utils_nlp/models/transformers/common.py

```diff
@@ -143,18 +148,22 @@ def fine_tune(
                 raise ImportError("Please install apex from https://www.github.com/nvidia/apex")
             self.model, optimizer = amp.initialize(self.model, optimizer, opt_level=fp16_opt_level)
 
-        # multi-gpu training (should be after apex fp16 initialization)
-        if n_gpu > 1:
-            self.model = torch.nn.DataParallel(self.model)
-
-        # Distributed training (should be after apex fp16 initialization)
         if local_rank != -1:
             self.model = torch.nn.parallel.DistributedDataParallel(
                 self.model,
                 device_ids=[local_rank],
                 output_device=local_rank,
                 find_unused_parameters=True,
             )
+        elif num_gpus > 1:
+            if not isinstance(self.model, torch.nn.DataParallel):
```

Comment:
There is still issue on this part. I specified only two GPUs for training and scoring but 4 GPUS was used.

Reply:
You were right, we had to pass the gpu list to DataParallel. I thought we we're doing that in get_device. It works now. Also thanks @hlums for cleaning it up and fixing another bug.

Reply:
We were able to reproduce your issue and fixed it. You can test it again when you are back in the office. Merging for now.